src/init.cpp

```diff
@@ -364,13 +364,8 @@ std::string HelpMessage(HelpMessageMode mode)
     strUsage += HelpMessageOpt("-prune=<n>", strprintf(_("Reduce storage requirements by pruning (deleting) old blocks. This mode disables wallet support and is incompatible with -txindex. "
             "Warning: Reverting this setting requires re-downloading the entire blockchain. "
             "(default: 0 = disable pruning blocks, >%u = target size in MiB to use for block files)"), MIN_DISK_SPACE_FOR_BLOCK_FILES / 1024 / 1024));
-#ifdef ENABLE_WALLET
-    strUsage += HelpMessageOpt("-reindex-chainstate", _("Rebuild chain state from the currently indexed blocks (implies -rescan unless pruning or unless -rescan=0 is explicitly specified)"));
-    strUsage += HelpMessageOpt("-reindex", _("Rebuild chain state and block index from the blk*.dat files on disk (implies -rescan unless pruning or unless -rescan=0 is explicitly specified)"));
-#else
```

Comment:
After talking with @daira and staring at the documentation of `-rescan`, I think it's reasonable for this documentation to remain (i.e. say that `-reindex` implies `-rescan` for the purpose of indicating to users that a `-rescan` won't help after a `-reindex`, but not soft-setting because that's misleading). This is because `-rescan` is documented as finding missing wallet transactions, and this will naturally occur as the chain data is replayed to the wallet during a `-reindex`.